contracts/common/Registry.sol

```diff
@@ -16,10 +15,16 @@ contract Registry is Ownable {
   mapping(bytes32 => address) contractMap;
   mapping(address => address) public rootToChildToken;
   mapping(address => address) public childToRootToken;
-  mapping(address => bool) public proofValidatorContracts;
   // @todo we can think of one function from the registry which returns both (childToken,isERC721) if we are using it frequently together.
+  mapping(address => bool) public proofValidatorContracts;
   mapping(address => bool) public isERC721;
 
+  enum Type { Invalid, ERC20, ERC721 }
+  struct Predicate {
+    Type _type;
```

Comment:
do we have plans for adding more meta-data into `Predicate` struct ?
else we can use `Type` in mapping right away.

Reply:
Yes, I was assuming we might need to add more predicate related data. But if and when that does happen, it would still require a code change in the struct so I could probably have `Type` directly in the mapping for now. Thoughts?

Reply:
yeah, sure for now let's keep it.